controllers/mirrorpeer_controller.go

```diff
@@ -225,6 +232,91 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return ctrl.Result{Requeue: true}, nil
 }
 
+func checkK8sErrors(ctx context.Context, err error, mirrorPeerCopy *multiclusterv1alpha1.MirrorPeer) (ctrl.Result, error) {
```

Comment:
can be transferred to utils/error.go since this is a common check.

Please check if this could be made generic for other resources as well and not just mirrorPeer